subsys/bluetooth/host/gatt.c

```diff
@@ -4989,6 +4998,14 @@ void bt_gatt_connected(struct bt_conn *conn)
 
 #if defined(CONFIG_BT_GATT_CLIENT)
 	add_subscriptions(conn);
+
+#if defined(CONFIG_BT_SETTINGS)
+	/* Register the gatt module for connection callbacks so it can be
+	 * notified when pairing has completed. This is used to enable CCC and
+	 * CF storage on pairing complete.
+	 */
+	bt_conn_cb_register(&gatt_conn_cb);
+#endif
```

Comment:
This is really the only #ifdef block that you should need for the callback struct:
```
#if defined(CONFIG_BT_SETTINGS)
        static struct bt_conn_cb gatt_conn_cb = {
                .identity_resolved = bt_gatt_identity_resolved;
        };

        bt_conn_cb_register(&gatt_conn_cb);
#endif

Reply:
Actually, you might even be able to use:
```c
if (IS_ENABLED(CONFIG_BT_SETTINGS) {
        static struct bt_conn_cb gatt_conn_cb = {
                .identity_resolved = bt_gatt_identity_resolved,
        };

        bt_conn_cb_register(&gatt_conn_cb);
}
```

That might let you skip the #ifdef block also for `bt_gatt_identity_resolved()` and any other static functions that it may call.

Reply:
Done.